gitpod-io/gitpod
# add ide settings permission and enable it for devops role

#### What it does

- fix #1948: `ide-settings` permission is added which is enabled by default for `devops` role. This permission allow to configure the default ide image as well as control workspace specific ide image from `.gitpod.yml`. Otherwise such settings are ignored.

#### How to test

- check then Theia is used even if .gitpod.yml contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com//#https://github.com/akosyakov/go-gin-app
- go to your settings page and check that `IDE Settings` section is not visible: https://ak-ide-permission.staging.gitpod-dev.com/settings/
- grant `devops` role to your user, i.e. run a query like: `update d_b_user set rolesOrPermissions = '["admin","devops"]' where name = 'akosyakov';`
- check that Code is used if .gitpod.yml contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com//#https://github.com/akosyakov/go-gin-app
- check that Theia is used if .gitpod.yml does NOT contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com/#https://github.com/gitpod-io/go-gin-app
- go to your settings page and switch the default ide to `Code` instead of `Theia`: https://ak-ide-permission.staging.gitpod-dev.com/settings/
- check that Code is used if .gitpod.yml does NOT contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com/#https://github.com/gitpod-io/go-gin-app
